File: cuecard-mobile/ios/CueCard/CueCard/Models/CueColor.swift
import SwiftUI

/// The color every delivery cue is drawn in.
///
/// One color covers the whole app — the choice lives in Settings, not in the
/// script — so changing it recolors every cue in every script at once.
///
/// The raw value is persisted with the settings, so it has to stay stable once
/// shipped: renaming a case resets that user's choice back to the default.
enum CueColor: String, Codable, CaseIterable, Identifiable {
    case pink
    case yellow
    case green
    case blue
    case purple
    case red

    var id: String { rawValue }

    /// What cues are drawn in until the user picks something else.
    static let `default`: CueColor = .pink

    var displayName: String {
        rawValue.capitalized
    }

    func color(for colorScheme: ColorScheme) -> Color {
        switch self {
        case .pink: return AppColors.pink(for: colorScheme)
        case .yellow: return AppColors.yellow(for: colorScheme)
        case .green: return AppColors.green(for: colorScheme)
        case .blue: return AppColors.blue(for: colorScheme)
        case .purple: return AppColors.purple(for: colorScheme)
        case .red: return AppColors.red(for: colorScheme)
        }
    }

    func uiColor(isDarkMode: Bool) -> UIColor {
        switch self {
        case .pink: return isDarkMode ? AppColors.UIColors.Dark.pink : AppColors.UIColors.Light.pink
        case .yellow: return isDarkMode ? AppColors.UIColors.Dark.yellow : AppColors.UIColors.Light.yellow
        case .green: return isDarkMode ? AppColors.UIColors.Dark.green : AppColors.UIColors.Light.green
        case .blue: return isDarkMode ? AppColors.UIColors.Dark.blue : AppColors.UIColors.Light.blue
        case .purple: return isDarkMode ? AppColors.UIColors.Dark.purple : AppColors.UIColors.Light.purple
        case .red: return isDarkMode ? AppColors.UIColors.Dark.red : AppColors.UIColors.Light.red
        }
    }
}

File: cuecard-mobile/android/app/src/main/java/com/thisisnsh/cuecard/android/Models/CueColor.kt
package com.thisisnsh.cuecard.android.models

import androidx.compose.ui.graphics.Color
import kotlinx.serialization.Serializable

/**
 * The color every delivery cue is drawn in.
 *
 * One color covers the whole app — the choice lives in Settings, not in the
 * script — so changing it recolors every cue in every script at once.
 *
 * The stored name is persisted with the settings, so it has to stay stable once
 * shipped: renaming a case resets that user's choice back to the default.
 */
@Serializable
enum class CueColor {
    PINK,
    YELLOW,
    GREEN,
    BLUE,
    PURPLE,
    RED;

    /** The persisted form, matching the iOS raw value exactly. */
    val rawValue: String
        get() = name.lowercase()

    val displayName: String
        get() = rawValue.replaceFirstChar { it.uppercase() }

    fun color(isDark: Boolean): Color = when (this) {
        PINK -> AppColors.pink(isDark)
        YELLOW -> AppColors.yellow(isDark)
        GREEN -> AppColors.green(isDark)
        BLUE -> AppColors.blue(isDark)
        PURPLE -> AppColors.purple(isDark)
        RED -> AppColors.red(isDark)
    }

    companion object {
        /** What cues are drawn in until the user picks something else. */
        val DEFAULT = PINK

        fun fromString(value: String?): CueColor =
            entries.find { it.rawValue == value?.lowercase() } ?: DEFAULT
    }
}
